### DISCUSSION TITLE
Next.js 13 font system and Chakra UI

### DISCUSSION THREAD

**Discussion by slava-lu** [vercel/next.js#44237] [Help] [answered]
Hi,
Thanks for the Tailwind CSS example. Now how do I use it with Chakra UI? Which leads to the point that your new font system is incompatible with any third party design system.  Can't you just generate the normal font families, not with random numbers attached??

**Comment by dextermb** [comment, marked as answer]
Does something like this work for you? Then you can simply use the variables in config.

```jsx
<style jsx global>
{`:root {
  --font-google-ibm-plex-sans: ${sans.style.fontFamily};
  --font-google-ibm-plex-mono: ${mono.style.fontFamily};
}`}
</style>
```

**Comment by slava-lu** [comment]
Thanks for the answer, but to be honest these 6 lines of code are a bit confusing for me.  
Is there any link where I can read what all this `:root`  thing is about?

**Reply by dextermb** [reply]
A quick Google would take you to MDN which is where I tend to read technical web related stuff. Here's a link to the `:root` documentation:

[developer.mozilla.org/en-US/docs/Web/CSS/:root](https://developer.mozilla.org/en-US/docs/Web/CSS/:root)

**Comment by slava-lu** [comment]
Thanks, It worked perfectly. 
Just in case someone will be looking for the same solution here is the example with Inter font as in the docs:

In `_app.js` add this code as [described in the doc](https://nextjs.org/docs/basic-features/font-optimization)
```
 <style jsx global>
        {`
          :root {
            --font-google-inter: ${inter.style.fontFamily};
          }
        `}
      </style>
```
then create the file `fonts.js`

```
const style = {
  heading: `var(--font-google-inter)`,
  body: `var(--font-google-inter)`,
  mono: `var(--font-google-inter)`,
}
export default style
```

and add it to the theme as [described here](https://chakra-ui.com/docs/styled-system/customize-theme)





```

**Reply by dextermb** [reply]
Would you be able to mark my response as the answer? Thanks

**Comment by jinsley8** [comment]
When I tested out the new Next.js 13 font system with Chakra UI I found that it was still slower and produced slightly larger filesize compared to the fontsource option here: https://chakra-ui.com/community/recipes/using-fonts#option-1-install-with-npm
